lib/webidl2.js

```diff
@@ -424,36 +424,24 @@
           consume(FLOAT) ||
           consume(INT) ||
           consume(STR);
-        if (!ret.rhs && consume(OTHER, "(")) {
+      }
+      all_ws();
+      if (consume(OTHER, "(")) {
+        if (eq && !ret.rhs) {
           // [Exposed=(Window,Worker)]
-          const rhs_list = [];
-          const id = consume(ID);
-          if (id) {
-            rhs_list.push(id.value);
-          }
-          identifiers(rhs_list);
-          consume(OTHER, ")") || error("Unexpected token in extended attribute argument list or type pair");
           ret.rhs = {
             type: "identifier-list",
-            value: rhs_list
+            value: identifiers()
           };
         }
-        if (!ret.rhs) return error("No right hand side to extended attribute assignment");
-      }
-      all_ws();
-      if (consume(OTHER, "(")) {
-        let args, pair;
-        // [Constructor(DOMString str)]
-        if (args = argument_list(store)) {
-          ret["arguments"] = args;
-        }
-        // [Constructor()]
         else {
-          ret["arguments"] = [];
+          // [NamedConstructor=Audio(DOMString src)] or [Constructor(DOMString str)]
+          ret.arguments = argument_list(store);
```

Comment:
I wonder if there was some old version of node that got confused by this. No need to change it back :)